File: Task-management/task-management-ui/src/services/projectService.js
import API from "./api";

export const getProjects = () =>
    API.get("/projects");

export const createProject = (project) =>
    API.post("/projects", project);

export const updateProject = (id, project) =>
    API.put(`/projects/${id}`, project);

export const deleteProject = (id) =>
    API.delete(`/projects/${id}`);

export const getProjectById = (id) =>
    API.get(`/projects/${id}`);

export const getProjectStats = () =>
    API.get("/projects/stats");

export const getHighPriorityProjects = () =>
    API.get("/projects/high-priority");

File: Task-management/task-management/src/main/java/com/taskmanagement/service/ProjectService.java
package com.taskmanagement.service;

import com.taskmanagement.dto.ProjectRequest;
import com.taskmanagement.dto.ProjectStatsResponse;
import com.taskmanagement.entity.Project;

import java.util.List;

public interface ProjectService {

    Project createProject(ProjectRequest request);

    List<Project> getAllProjects();

    Project getProject(Long id);

    Project updateProject(Long id, ProjectRequest request);

    void deleteProject(Long id);

    ProjectStatsResponse getProjectStats();

    // NEW: Get all high priority projects
    List<Project> getHighPriorityProjects();
}